code/modules/mob/dead/new_player/latejoin_menu.dm

```diff
@@ -0,0 +1,192 @@
+
+	if(!input_contents)
+		return
+
+	user.AttemptLateSpawn(input_contents)
+
+/datum/latejoin_menu/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "JobSelection", "Latejoin Menu")
+		ui.open()
+
+/datum/latejoin_menu/ui_data(mob/user)
+	var/mob/dead/new_player/owner = user
+	var/list/departments = list()
+	var/list/data = list(
+		"disable_jobs_for_non_observers" = SSlag_switch.measures[DISABLE_NON_OBSJOBS],
+		"round_duration" = DisplayTimeText(world.time - SSticker.round_start_time),
+		"departments" = departments,
+	)
+	if(SSshuttle.emergency)
+		switch(SSshuttle.emergency.mode)
+			if(SHUTTLE_ESCAPE)
+				data["shuttle_status"] = "The station has been evacuated."
+			if(SHUTTLE_CALL)
+				if(!SSshuttle.canRecall())
+					data["shuttle_status"] = "The station is currently undergoing evacuation procedures."
+
+	for(var/datum/job/prioritized_job in SSjob.prioritized_jobs)
+		if(prioritized_job.current_positions >= prioritized_job.total_positions)
+			SSjob.prioritized_jobs -= prioritized_job
+
+	for(var/datum/job_department/department as anything in SSjob.joinable_departments)
+		var/list/department_jobs = list()
+		var/list/department_data = list(
+			"jobs" = department_jobs,
+			"open_slots" = 0,
+		)
+		departments[department.department_name] = department_data
+
+		for(var/datum/job/job_datum as anything in department.department_jobs)
+			var/job_availability = owner.IsJobUnavailable(job_datum.title, latejoin = TRUE)
+
+			var/list/job_data = list(
+				"prioritized" = (job_datum in SSjob.prioritized_jobs),
+				"used_slots" = job_datum.current_positions,
+				"open_slots" = job_datum.total_positions < 0 ? "∞" : job_datum.total_positions,
+			)
+
+			if(job_availability != JOB_AVAILABLE)
+				job_data["unavailable_reason"] = get_job_unavailable_error_message(job_availability, job_datum.title)
+
+			if(job_datum.total_positions < 0)
+				department_data["open_slots"] = "∞"
+
+			if(department_data["open_slots"] != "∞")
+				department_data["open_slots"] += job_datum.total_positions - job_datum.current_positions
+
+			department_jobs[job_datum.title] = job_data
+
+	return data
+
+/datum/latejoin_menu/ui_static_data(mob/user)
+	var/list/departments = list()
+
+	for(var/datum/job_department/department as anything in SSjob.joinable_departments)
+		var/list/department_jobs = list()
+		var/list/department_data = list(
+			"jobs" = department_jobs,
+			"color" = department.ui_color,
+		)
+		departments[department.department_name] = department_data
+
+		for(var/datum/job/job_datum as anything in department.department_jobs)
+			var/datum/outfit/outfit = job_datum.outfit
+			var/datum/id_trim/trim = initial(outfit.id_trim)
+
+			var/list/job_data = list(
+				"command" = !!(job_datum.departments_bitflags & DEPARTMENT_BITFLAG_COMMAND),
+				"description" = job_datum.description,
+				"icon" = initial(trim.orbit_icon)
```

Comment:
```suggestion
				"icon" = initial(trim.orbit_icon),
```

Nit: These are likely to be expanded upon by someone else.

Reply:
I usually add commas at the end of lists- must've forgotten!